dot/core/service.go

```diff
@@ -474,20 +475,17 @@ func (s *Service) handleChainReorg(prev, curr common.Hash) error {
 }
 
 func (s *Service) notifyRuntimeUpdated(version runtime.Version) {
-	s.runtimeChangedLock.RLock()
-	defer s.runtimeChangedLock.RUnlock()
+	s.runtimeUpdateSubscriptionsLock.RLock()
+	defer s.runtimeUpdateSubscriptionsLock.RUnlock()
 
-	if len(s.runtimeChanged) == 0 {
+	if len(s.runtimeUpdateSubscriptions) == 0 {
 		return
 	}
 
-	logger.Info("notifying runtime updated chans...", "chans", s.runtimeChanged)
-	for _, ch := range s.runtimeChanged {
+	logger.Info("notifying runtime updated chans...", "chans", s.runtimeUpdateSubscriptions)
+	for _, ch := range s.runtimeUpdateSubscriptions {
 		go func(ch chan<- runtime.Version) {
-			select {
-			case ch <- version:
-			default:
-			}
+			ch <- version
```

Comment:
what if someone unregisters/closes the channel at the same time this happens? I would assume it panics.

Reply:
Any suggestions regarding how I should test this, and/or fixes?

Reply:
you could call `notifyRuntimeUpdated` and in parallel goroutines call `Register/UnregisterRuntimeUpdatedChannel`.  You should be able to produce a panic.  There are a few ways to fix it, but let's see if we can get it to panic first.

Reply:
yeah probably related to #1701 , you could add a per-channel lock and `closed bool` as a potential solution, @timwu20  do you know of any other good sulutions?

Reply:
wait nvm, you lock the whole map before closing any of the channels. I think that should fix it